python/beeai_framework/tools/types.py

```diff
@@ -56,9 +64,9 @@ def get_text_content(self) -> str:
         return self.result
 
 
-class JSONToolOutput(ToolOutput):
-    def __init__(self, result: Any) -> None:
-        self.result = result
+class JSONToolOutput(Generic[T], ToolOutput[T]):
+    def __init__(self, result: T, **kwargs: Any) -> None:
+        super().__init__(result=result, **kwargs)
```

Comment:
The constructor is duplicated to prevent getting complaints from the Mypy Plugin in the IDE